### ISSUE TITLE
Improve Dialog Accessibility

### ISSUE THREAD

**Issue by joechrisman** [primefaces/primeng#4161]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**
Screen readers have trouble reading the dialog because it is missing several aria/role attributes

**Expected behavior**
Dialog should function appropriately with screen readers

* **Angular version:** 2.0.X
4.3.5

* **PrimeNG version:** 2.0.X
4.2.2

* **Browser:** 
Chrome 61
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript 2.2

**Comment by cagataycivici** [comment]
Added role="dialog" and ariaLabelledBy property as well to provide the id of the header element that maps to aria-labelledby of dialog. So defining ariaLabelledBy is suggested.
